lib/internal/managers_for_classes.dart

```diff
@@ -42,7 +42,7 @@ abstract class ManagerForPeripheral {
       bool withResponse,
       String transactionId);
 
-  Stream<CharacteristicWithValue> monitorCharacteristicForDevice(
+  Stream<CharacteristicWithValueWithTransactionId> monitorCharacteristicForDevice(
```

Comment:
Also: transactionId doesn't need to be visible to the user, it's an internal mechanism of the library.